File: app/src/main/java/com/intetm/comicbreeze/service/database/DatabaseService.java
package com.intetm.comicbreeze.service.database;

import android.arch.persistence.room.Room;
import android.content.Context;

public class DatabaseService {

    private static DatabaseService INSTANCE  = null;
    public static DatabaseService getInstance() {
      return INSTANCE;
    }
    public static void init(Context applicationContext) {
        INSTANCE = new DatabaseService(applicationContext);
    }

    public final AppDatabase db;

    private DatabaseService(Context applicationContext) {
         db = Room.databaseBuilder(applicationContext, AppDatabase.class, "db").build();
    }
}

File: app/src/main/java/com/intetm/comicbreeze/service/database/DatabaseService.kt
package com.intetm.comicbreeze.service.database

import android.arch.persistence.room.Room
import android.content.Context

class DatabaseService private constructor(applicationContext: Context) {

    val db: AppDatabase = Room.databaseBuilder<AppDatabase>(applicationContext, AppDatabase::class.java, "db").build()

    companion object {

        var instance: DatabaseService? = null
            private set

        fun init(applicationContext: Context) {
            instance = DatabaseService(applicationContext)
        }
    }
}
